### PR TITLE
first commit for getting weave onto k2

### PR DESCRIPTION
**Pull request by alejandroEsc**
may need some more work? going to test it on k8s 1.6 when @coffeepac says this is stable. Hoping that means today. ref: https://github.com/samsung-cnct/k2/issues/485

### CODE REVIEW

File: ansible/roles/kraken.fabric/kraken.fabric.weave/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+Role Name
```
Comment (davidewatson) [inline ansible/roles/kraken.fabric/kraken.fabric.weave/README.md]: I know you didn't start this trend, but please do not add new empty placeholders (unless of course you also open a ticket and also reference it). In particular, these role level README.md's appear to be mostly (all?) empty.

Comment (alejandroEsc) [inline ansible/roles/kraken.fabric/kraken.fabric.weave/README.md]: As you say, just following convention, note that the second part of this is to update the readmefiles. Should i remove these where i find them? would avoid confusion in the future specially since this seems to be establishing our style.

File: ansible/roles/kraken.config/files/gke-config.yaml [hunk 2]
```diff
@@ -83,6 +83,38 @@ definitions:
         default: *defaultCanalFabric
         versions:
           v1.6: *defaultCanalFabric16
+    - &defaultWeaveFabric
```
Comment (alejandroEsc) [inline ansible/roles/kraken.config/files/gke-config.yaml]: gotcha, will do so.

File: ansible/roles/kraken.fabric/kraken.fabric.weave/README.md (hunk 1 above)
Comment (davidewatson) [inline ansible/roles/kraken.fabric/kraken.fabric.weave/README.md]: It should probably be a separate PR, but yeah please do, useless documentation is usually worse than no documentation.

**Comment by coffeepac** [comment]
1.6 is stable.  go nuts.  its not the default yet (#361).  that should be tomorrow or next week.

**Comment by alejandroEsc** [comment]
test this please

**Comment by alejandroEsc** [comment]
@coffeepac i think this build failed because of tiller? anything i can do to either help get that fixed, or skip that test? I have seen this issue in 1.6 k8s only.

**Comment by alejandroEsc** [comment]
@davidewatson this is available for reviewing again. thank you!

File: schemas/config/v1/weaveFabricConfig.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (davidewatson) [inline schemas/config/v1/weaveFabricConfig.json]: This shouldn't pass CI because it should fail the validation step in ansible. In order to use the new schema you have introduced, you need to update schemas/config/v1/fabricConfig.json. Each property also needs a title, documentation and required fields.

Comment (alejandroEsc) [inline schemas/config/v1/weaveFabricConfig.json]: ohhh where could i see that failure? clearly i missed it somewhere?

Comment (davidewatson) [inline schemas/config/v1/weaveFabricConfig.json]: The latest job which ran failed [here](https://common-jenkins.kubeme.io/job/K2/job/PR-486/17/console) and must be restarted.

File: schemas/config/v1/weaveFabricContainers.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "id": "http://judkins.house/apis/k2/v1/weaveFabricContainers.json",
+
+  "description": "Weave Fabric Container Options configuration",
+  "properties": {
+    "weave": { "$ref": "http://judkins.house/apis/k2/v1/weaveFabricContainerConfig.json" },
```
Comment (davidewatson) [inline schemas/config/v1/weaveFabricContainers.json]: Recently $refs were changed to be relative (e.g. `+    "weave": { "$ref": "weaveFabricContainerConfig.json" } `). This is necessary for some of the tools we are using.

File: schemas/config/v1/fabricConfig.json [hunk 5]
```diff
@@ -7,6 +7,7 @@
   "properties": {
     "oneOf": [
       { "$ref": "canalFabricConfig.json" },
+      { "$ref": "WeaveFabricConfig.json" },
```
Comment (davidewatson) [inline schemas/config/v1/fabricConfig.json]: s/Weave/weave/g

File: schemas/config/v1/weaveFabricContainers.json (hunk 4 above)
Comment (alejandroEsc) [inline schemas/config/v1/weaveFabricContainers.json]: understood, changing all affected items.

File: ansible/roles/kraken.nodePool/kraken.master.docker/templates/v1.6/controller-manager-manifest.yaml.jinja2 [hunk 6]
```diff
@@ -16,8 +16,10 @@ spec:
     - --cloud-provider={{ cluster.providerConfig.provider }}
     - --service-account-private-key-file=/etc/kubernetes/ssl/service-account-key.pem
     - --root-ca-file=/etc/kubernetes/ssl/ca.pem
-    - --cluster-cidr={{ fabricConfigVal.options.network.network }}
```
Comment (davidewatson) [inline ansible/roles/kraken.nodePool/kraken.master.docker/templates/v1.6/controller-manager-manifest.yaml.jinja2]: LGTM, pending a change to this template logic.

File: ansible/roles/kraken.nodePool/kraken.master.docker/templates/v1.6/controller-manager-manifest.yaml.jinja2 [hunk 7]
```diff
@@ -16,8 +16,10 @@ spec:
...
+{% if fabricConfigVal.options.network is defined %}
```
Comment (davidewatson) [inline ansible/roles/kraken.nodePool/kraken.master.docker/templates/v1.6/controller-manager-manifest.yaml.jinja2]: LGTM, pending a change to this template logic.
